core/sdk-core/src/main/java/software/amazon/awssdk/core/interceptor/Context.java

```diff
@@ -192,14 +192,16 @@ public interface FailedExecution {
         SdkRequest request();
 
         /**
-         * The latest version of the {@link SdkHttpFullRequest} available when the execution failed. If the execution failed
-         * before or during request marshalling, this will return {@link Optional#empty()}.
+         * The latest version of the {@link SdkHttpFullRequest} available when the execution failed. To access the
+         * {@link SdkHttpFullRequest}, cast the returned {@link SdkHttpRequest}. If the execution failed before or
+         * during request marshalling, this will return {@link Optional#empty()}.
```

Comment:
> {@link SdkHttpFullRequest}, cast the returned {@link SdkHttpRequest}.

 This makes it sound like `SdkHttpRequest` is always a `SdkHttpFullRequest` which may technically not be true in the future. Can you add something along the lines of it *may* be `SdkHttpFullRequest`, and if so they can cast it to get access to the additional data?

Reply:
clarified